src/gui/speedlimitdialog.cpp

```diff
@@ -50,58 +64,114 @@ SpeedLimitDialog::~SpeedLimitDialog()
+void SpeedLimitDialog::updateAltDownloadSpinValue(const int value)
+{
+    m_ui->spinAltDownloadLimit->setValue(value);
+}
 
-    if (dlg.exec() == QDialog::Accepted)
-    {
-        if (ok) *ok = true;
+void SpeedLimitDialog::updateAltUploadSpinValue(const int value)
+{
+    m_ui->spinAltUploadLimit->setValue(value);
+}
 
-        const int val = dlg.getSpeedLimit();
-        if (val < 0)
-            return 0;
-        return (val * 1024);
-    }
+void SpeedLimitDialog::updateDownloadSliderValue(const int value)
+{
+    if (value > m_ui->sliderDownloadLimit->maximum())
+        m_ui->sliderDownloadLimit->setMaximum(value);
+    m_ui->sliderDownloadLimit->setValue(value);
+}
 
-    return -2;
+void SpeedLimitDialog::updateUploadSliderValue(const int value)
+{
+    if (value > m_ui->sliderUploadLimit->maximum())
+        m_ui->sliderUploadLimit->setMaximum(value);
+    m_ui->sliderUploadLimit->setValue(value);
 }
 
-void SpeedLimitDialog::updateSpinValue(const int value)
+void SpeedLimitDialog::updateAltDownloadSliderValue(const int value)
 {
-    m_ui->spinBandwidth->setValue(value);
+    if (value > m_ui->sliderAltDownloadLimit->maximum())
+        m_ui->sliderAltDownloadLimit->setMaximum(value);
+    m_ui->sliderAltDownloadLimit->setValue(value);
 }
 
-void SpeedLimitDialog::updateSliderValue(const int value)
+void SpeedLimitDialog::updateAltUploadSliderValue(const int value)
 {
-    if (value > m_ui->bandwidthSlider->maximum())
-        m_ui->bandwidthSlider->setMaximum(value);
-    m_ui->bandwidthSlider->setValue(value);
+    if (value > m_ui->sliderAltUploadLimit->maximum())
+        m_ui->sliderAltUploadLimit->setMaximum(value);
+    m_ui->sliderAltUploadLimit->setValue(value);
 }
 
-int SpeedLimitDialog::getSpeedLimit() const
+int SpeedLimitDialog::getDownloadSpeedLimit() const
 {
-    return m_ui->spinBandwidth->value();
+    return m_ui->spinDownloadLimit->value();
 }
 
-void SpeedLimitDialog::setupDialog(long maxSlider, long val)
+int SpeedLimitDialog::getUploadSpeedLimit() const
 {
-    val = qMax<long>(0, val);
+    return m_ui->spinUploadLimit->value();
+}
 
-    if (maxSlider <= 0)
-        maxSlider = 10000;
+int SpeedLimitDialog::getAltDownloadSpeedLimit() const
+{
+    return m_ui->spinAltDownloadLimit->value();
+}
 
-    // This can happen for example if global rate limit is lower
-    // than torrent rate limit.
-    if (val > maxSlider)
-        maxSlider = val;
+int SpeedLimitDialog::getAltUploadSpeedLimit() const
+{
+    return m_ui->spinAltUploadLimit->value();
+}
 
-    m_ui->bandwidthSlider->setMaximum(maxSlider);
-    m_ui->bandwidthSlider->setValue(val);
-    updateSpinValue(val);
+void SpeedLimitDialog::initDialog()
+{
+    m_ui->labelGlobalSpeedIcon->setPixmap(Utils::Gui::scaledPixmap(UIThemeManager::instance()->getIcon(QLatin1String("slow_off")), this, 16));
+    m_ui->labelAltGlobalSpeedIcon->setPixmap(Utils::Gui::scaledPixmap(UIThemeManager::instance()->getIcon(QLatin1String("slow")), this, 16));
```

Comment:
I would move these two lines in constructor and keep only limit-related assignments here. Otherwise, I don't understand based on what concept you extracted this code from the constructor.

Reply:
Can't see why it matters but sure.

Reply:
>Can't see why it matters but sure.

You have some initialization code in the constructor directly and some other is separated into this method. I'm sorry, but I didn't see any pattern on which it is based (maybe I just didn't understand it). But then what's the point? Why not put all the code in the constructor? Or Vice versa, in this method?

Reply:
>Why not put all the code in the constructor? Or Vice versa, in this method?

No particular reason. It was this way initially in this dialog and i just continued it. Want me to move everything in the constructor? It will have to be a new commit though because i'm not wasting my time resolving conflicts between existing commits or trying to figure out what goes where.

Reply:
>It will have to be a new commit though because i'm not wasting my time resolving conflicts between existing commits or trying to figure out what goes where.

It contradicts our rules. We do not merge PRs that have a dirty history. You will have to apply any required changes to the corresponding commits. Or just squash all your commits into one in the end (even now, it doesn't look like all of them are independent enough).

Reply:
>Or just squash all your commits into one in the end (even now, it doesn't look like all of them are independent enough).

Thank god. I'll squash the last 3 for "Torrent options" at least and we'll see from there.

>Want me to move everything in the constructor?

Well?

Reply:
>>Want me to move everything in the constructor?
>>
>Well?

I want this to be a logical separation, if necessary (for example, you can extract the initialization code for control values from the current Session property values, and leave everything else in the constructor). Or you can extract the code that is called from several places (but this is not the case here).
If in doubt, just put everything in the constructor, because the current split is only confusing.

Reply:
I moved everything.